src/sdk/namespace_fs.js

```diff
@@ -537,6 +537,25 @@ class NamespaceFS {
         }
     }
 
+    /**
+     * _should_update_issues_report is intended to avoid updating the namespace issues report in case:
+     * 1. The key doesn't exist and the path is not internal -
+     *    internal path is created for specific cases, for example in version.
+     *    Note: it also covers the delete marker case (since it is in a versioned path)
+     * @param {object} params
+     * @param {string} file_path
+     * @param {object} err
+     */
+    _should_update_issues_report(params, file_path, err) {
+        let res = true;
+        const { key } = params;
+        const md_file_path = this._get_file_md_path({ key });
+        const non_internal_path = file_path === md_file_path;
+        const no_such_key_condition = err.code === `ENOENT` && non_internal_path;
+        if (no_such_key_condition) res = false;
+        return res;
+    }
```

Comment:
```suggestion
    _should_update_issues_report(params, file_path, err) {
        const { key } = params;
        const md_file_path = this._get_file_md_path({ key });
        const non_internal_path = file_path === md_file_path;
        const no_such_key_condition = err.code === `ENOENT` && non_internal_path;
        return !no_such_key_condition;
    }
```

Reply:
I can change it, the reason I tried to write it this way was to avoid negative conditions and easily expand this for other cases.